### PR TITLE
admin search results: give a button to reset google ID of a student #2919

### PR DESCRIPTION
**Pull request by ymoctavia**
for #2919

### CODE REVIEW

File: src/main/java/teammates/common/util/Const.java [hunk 1]
```diff
@@ -869,6 +870,9 @@
         public static final String JOIN_COURSE_GOOGLE_ID_BELONGS_TO_DIFFERENT_USER = "The Google ID %s belongs to an existing user in the course."
                 + "Please login again using a different Google account, and try to join the course again.";
         
+        public static final String STUDENT_GOOGLEID_RESET = "The student's google id has been reset";
+        public static final String STUDENT_GOOGLEID_RESET_FAIL = "Error occurs when trying to reset student's google id";
```
Comment (damithc) [inline src/main/java/teammates/common/util/Const.java]: An error occurred when ...

**Comment by ymoctavia** [comment]
Professor @damithc , now

1.  original student home page with old google id


2. instructor's view of joining status:


3.admin search for this student:


4.hit the reset button:


5. google id is reset to null




6. instructor's view of joining status after reset:


7.student re-joint using new google id:


8.previous submission still exist:


9.admin's view of  this student in a new search


10. instructor's view of joining status after rejoining:

File: src/main/java/teammates/ui/controller/AdminSearchPageAction.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import teammates.common.datatransfer.InstructorAttributes;
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/AdminSearchPageAction.java]: Can we avoid using AdminSearchPageAction for this? Resetting Google ID is not related to search, although it appears in the search result page. Define a new action?

Comment (ymoctavia) [inline src/main/java/teammates/ui/controller/AdminSearchPageAction.java]: yes it should be done in an new action. Moreover, should we delete the original account entity if the account has no related student entity anymore after the reset (which should not be deleted if the account has already been registered with other "student"s )?

Comment (damithc) [inline src/main/java/teammates/ui/controller/AdminSearchPageAction.java]: Ideally, the account should be deleted if it doesn't have any other student
or instructor objects. But no big deal if it is not deleted.

**Comment by ymoctavia** [comment]
professor @damithc ,
here is the screenshot for re-join email:
capture

**Comment by damithc** [comment]
Rephrase first sentence : As requested, we have reset the link from your
TEAMMATES account to its previous Google account.
>
>

**Comment by ymoctavia** [comment]
Professor @damithc , I rephrased and added the "course" because this student's google id in the specific "course" is reset.

File: src/main/java/teammates/ui/controller/AdminStudentGoogleIdResetAction.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package teammates.ui.controller;
+
+import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Const;
+import teammates.logic.api.GateKeeper;
+import teammates.logic.api.Logic;
+
+public class AdminStudentGoogleIdResetAction extends Action {
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/AdminStudentGoogleIdResetAction.java]: Add a header comment for the class

File: src/main/java/teammates/ui/controller/AdminStudentGoogleIdResetAction.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    
+    private void deleteAccountIfNeeded(String wrongGoogleId){
+        Logic logic = new Logic();
+        
+        if(logic.getStudentsForGoogleId(wrongGoogleId).isEmpty()
+           && logic.getInstructorsForGoogleId(wrongGoogleId).isEmpty()){
+            logic.deleteAccount(wrongGoogleId);
+        }
+        
+        System.out.print("**************");
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/AdminStudentGoogleIdResetAction.java]: ???

**Comment by damithc** [comment]
Can merge after addressing the comments

**Pull request merged**